test/numbers.jl

```diff
@@ -2641,6 +2641,8 @@ end
     @test !isone(triu(fill(1, 5, 5)))
     @test !isone(zeros(Int, 5, 5))
     @test isone(Matrix(1I, 5, 5))
+    @test !isone(view(rand(5,5), [1,3,4], :))
+    @test isone(view(Diagonal([1,1, 1]), [1,2], 1:2))
```

Comment:
It appears that this was working before, but it was doing effectively `x == one(x)`, which created a whole dense copy in memory. So the new feature with this PR is that in most cases (except for things like `AbstractQ` from the `qr`s) the `isone`-check is now allocation-free. So we should actually test that as well:
```suggestion
    Dv = view(Diagonal([1,1, 1]), [1,2], 1:2)
    @test isone(Dv)
    @test (@allocated isone(Dv)) == 0
```
The last test fails currently.